spring-native/src/main/java/org/springframework/aot/test/AotCacheAwareContextLoaderDelegate.java

```diff
@@ -38,22 +42,69 @@
 
 	private final AotContextLoader aotContextLoader;
 
+	public AotCacheAwareContextLoaderDelegate() {
+		this(new AotContextLoader());
+	}
+
 	AotCacheAwareContextLoaderDelegate(AotContextLoader aotContextLoader) {
 		this.aotContextLoader = aotContextLoader;
 	}
 
-	public AotCacheAwareContextLoaderDelegate() {
-		this(new AotContextLoader());
+	AotCacheAwareContextLoaderDelegate(AotContextLoader aotContextLoader, ContextCache contextCache) {
+		super(contextCache);
+		this.aotContextLoader = aotContextLoader;
 	}
 
 	@Override
 	protected ApplicationContext loadContextInternal(MergedContextConfiguration config) throws Exception {
 		SmartContextLoader contextLoader = this.aotContextLoader.getContextLoader(config.getTestClass());
 		if (contextLoader != null) {
-			logger.info("Starting test in AOT mode using " + contextLoader);
-			return contextLoader.loadContext(config);
+			Assert.isInstanceOf(AotMergedContextConfiguration.class, config);
```

Comment:
These two lines are unrelated to the task here, isn't it?

Reply:
The log statement has been handled separately in 4787862c319fc609821922102f41e991abb5b1ef.

The other line is related to this PR.